server.js

```diff
@@ -0,0 +1,105 @@
+// TASK 2
+import express from 'express'
+import Student from './models/Student.js'
+import './db_connect.js'
+import createError from 'http-errors'
+
+const app = express()
+
+app.use(express.json())
+
+app.get("/", (req, res)=> {
+    res.send(`<h1>Hello from Students API</h1>
+    <a href="/students">/users</a>`)
+})
+
+app.get("/students", async (req, res)=> {
+    const students = await Student.find() 
+    res.json(students)
+})
+
+app.get("/students/:id", async (req, res, next) => {
+    const {id} = req.params
+    console.log(id);
+    try {
+        const student = await Student.findById(id)
+        if(!student){
+            throw new createError(
+                400,
+                "This ID doesn't exist"
+            )
+        }
+        res.json(student)
+    } 
+    catch (err) {
+        next(err)
+    }
+})
+
+app.post("/students", async (req, res, next)=> {
+    console.log(req.body);
+    try {
+        const newStudent = await Student.create(req.body)
+        res.json(newStudent)
+    } 
+    catch (err) {
+        next(err)
+    }
+})
+
+app.patch("/students/:id", async (req, res, next)=> {
+    const { id } = req.params;
+    console.log(id);
+    const studentData = req.body
+    console.log(studentData);
+    try {
+        const studentUpdated = await Student.findByIdAndUpdate(
+            id, 
+            studentData,
+            { new: true }
+            )
+            if(!studentUpdated){
```

Comment:
Ouch,  that try block code indentation hurts a bit the eyes :) But I can still read it somehow :)